Drivers: UART pdl-based driver bug fixed

aditi-bhaskar opened the pull request:
- Resolved failure in uart_basic_api test (rx now handled)
- Fixed interrupt mask configuration in uart-pdl driver
- Removed redundant irq handling functions

sreeramIfx commented on drivers/serial/uart_ifx_cat1_pdl.c:
```diff
@@ -67,7 +59,6 @@ struct ifx_cat1_uart_config {
 	uint16_t irq_num;
 	uint8_t irq_priority;
-	cy_cb_scb_uart_handle_events_t uart_handle_events_func;
 	void (*irq_config_func)(const struct device *dev);
```
I think this pointer is not being used in the driver. Should be deleted.

sreeramIfx commented on drivers/serial/uart_ifx_cat1_pdl.c:
```diff
@@ -389,18 +369,47 @@ static int ifx_cat1_uart_fifo_read(const struct device *dev, uint8_t *rx_data, c
+	if (event & CY_SCB_UART_TRANSMIT_EMTPY) {
+		tx_mask |= CY_SCB_UART_TX_EMPTY;
+	}
+	if (event & CY_SCB_UART_TRANSMIT_ERR_EVENT) {
```
```suggestion
	
	if (event & CY_SCB_UART_TRANSMIT_ERR_EVENT) {
```

sreeramIfx commented on drivers/serial/uart_ifx_cat1_pdl.c:
```diff
@@ -389,18 +369,47 @@ static int ifx_cat1_uart_fifo_read(const struct device *dev, uint8_t *rx_data, c
-	uint32_t rx_mask = 0;
+	NVIC_ClearPendingIRQ(config->irq_num);
+
+	uint32_t tx_mask = 0x0;
+	uint32_t rx_mask = 0x0;
```
Declarations should be grouped at the beginning of the function, please also update this across the file

sreeramIfx commented on drivers/serial/uart_ifx_cat1_pdl.c:
```diff
@@ -389,18 +369,47 @@ static int ifx_cat1_uart_fifo_read(const struct device *dev, uint8_t *rx_data, c
+	if (event & CY_SCB_UART_RECEIVE_NOT_EMTPY) {
+		rx_mask |= CY_SCB_UART_RX_NOT_EMPTY;
+	}
+	if (event & CY_SCB_UART_RECEIVE_ERR_EVENT) {
```
```suggestion

	if (event & CY_SCB_UART_RECEIVE_ERR_EVENT) {
```

sreeramIfx commented on drivers/serial/uart_ifx_cat1_pdl.c:
```diff
@@ -578,10 +535,36 @@ static void ifx_cat1_uart_irq_callback_set(const struct device *dev,
+	/* Clear interrupt */
+	CySCB_Type *base = config->reg_addr;
+	uint32_t locRxErr = (CY_SCB_UART_RECEIVE_ERR & Cy_SCB_GetRxInterruptStatusMasked(base));
+	uint32_t rx_clear = locRxErr | CY_SCB_RX_INTR_UART_BREAK_DETECT | CY_SCB_RX_INTR_LEVEL |
```
The additional events don't seem to be enabled in this file. Are these enabled elsewhere?

sreeramIfx commented on drivers/serial/uart_ifx_cat1_pdl.c:
```diff
@@ -918,6 +807,25 @@ static DEVICE_API(uart, ifx_cat1_uart_driver_api) = {
+			    uart_handle_events_func_##n, DEVICE_DT_INST_GET(n), 0);                \
+	}
+#else
+#define INTERRUPT_DRIVEN_UART_INIT(n)                                                              \
+	static void ifx_cat1_uart_irq_config_func_##n(const struct device *dev)                    \
+	{                                                                                          \
+	}
```
```suggestion
#define INTERRUPT_DRIVEN_UART_INIT(n)                                                              \
```

aditi-bhaskar committed:
Drivers: UART pdl-based driver bug fixed

 - Resolved failure in uart_basic_api test (rx now handled)
 - Fixed interrupt mask configuration in uart-pdl driver
 - Removed redundant irq handling functions

Signed-off-by: Aditi Bhaskar <aditi.bhaskar@infineon.com>
